HashRegistrarSimplified.sol

```diff
@@ -0,0 +1,367 @@
+    event OwnerChanged(address newOwner);
+    event DeedClosed();
+    bool active;
+
+    modifier onlyRegistrar {
+        if (msg.sender != address(registrar)) throw;
+        _
+    }
+
+    modifier onlyActive {
+        if (!active) throw;
+        _
+    }
+    
+    function Deed() {
+        registrar = Registrar(msg.sender);
+        creationDate = now;
+        active = true;
+    }
+        
+    function setOwner(address newOwner) onlyRegistrar {
+        owner = newOwner;
+        OwnerChanged(newOwner);
+    }
+    
+    function setBalance(uint newValue) onlyRegistrar onlyActive {
+        // Check if it has enough balance to set the value
+        if (this.balance < newValue) throw;
+        // Send the difference to the owner
+        if (!owner.send(this.balance - newValue)) throw;
+    }
+
+    function closeDeed(uint refundRatio) onlyRegistrar onlyActive {
+        active = false;            
+        if (! burn.send(((1000 - refundRatio) * this.balance)/1000)) throw;
+        DeedClosed();
+        destroyDeed();
+    }    
+
+    function destroyDeed() {
+        if (active) throw;
+        if(owner.send(this.balance)) 
+            selfdestruct(burn);
+    }
+
+    /* The default function just receives an amount */
+    function () {}
+}
+
+contract Registrar {
+    ENS public ens;
+    bytes32 public rootNode;
+
+    mapping (bytes32 => entry) public entries;
+    mapping (bytes32 => Deed) public sealedBids;
+    
+    enum Mode { Open, Auction, Owned, Forbidden }
+    uint32 constant auctionLength = 7 days;
+    uint32 constant revealPeriod = 24 hours;
+    uint constant minPrice = 0.01 ether;
+    uint public registryCreated;
+
+    event AuctionStarted(bytes32 hash, uint auctionExpiryDate);
+    event NewBid(bytes32 hash, uint deposit);
+    event BidRevealed(bytes32 hash, address owner, uint value, uint8 status);
+    event HashRegistered(bytes32 hash, address owner, uint value, uint now);
+    event HashReleased(bytes32 hash, uint value);
+    event HashInvalidated(bytes32 hash, string name, uint value, uint now);
+
+    struct entry {
+        Mode status;
+        Deed deed;
+        uint registrationDate;
+        uint value;
+        uint highestBid;
+    }
+    
+    modifier noEther {
+        if (msg.value > 0) throw;
+        _
+    }
+
+    modifier onlyOwner(bytes32 _hash) {
+        entry h = entries[_hash];
+        if (msg.sender != h.deed.owner() || h.status != Mode.Owned) throw;
+        _
+    }
+    
+    function Registrar(address _ens, bytes32 _rootNode) noEther {
+        ens = ENS(_ens);
+        rootNode = _rootNode;
+
+        registryCreated = now;
+    }
+
+    function max(uint a, uint b) internal constant returns (uint max) {
+        if (a > b)
+            return a;
+        else
+            return b;
+    }
+
+    function  min(uint a, uint b) internal constant returns (uint min) {
+        if (a < b)
+            return a;
+        else
+            return b;
+    }
+
+    /*
```

Comment:
Then I didn't understand how it works because the len() function takes a slice, not a string 

> On Sep 13, 2016, at 19:12, Nick Johnson <notifications@github.com> wrote:
> 
> In HashRegistrarSimplified.sol:
> 
> > +
> > +    function max(uint a, uint b) internal constant returns (uint max) {
> > +        if (a > b)
> > +            return a;
> > +        else
> > +            return b;
> > +    }
> > +
> > +    function  min(uint a, uint b) internal constant returns (uint min) {
> > +        if (a < b)
> > +            return a;
> > +        else
> > +            return b;
> > +    }
> > +
> > +    /*
> I would actually do away with the slice entirely, and just count characters on the actual string.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>